src/parse/stmt.rs

```diff
@@ -441,7 +439,8 @@ mod tests {
     fn parse_stmt(stmt: &str) -> CompileResult<Option<Stmt>> {
         let mut p = parser(stmt);
         let exp = p.statement();
-        if let Some(Err(err)) = p.pending.pop_front() {
+        if !p.error_handler.is_empty() {
+            let err = p.error_handler.into_iter().next().unwrap();
```

Comment:
Not sure this has the same semantics but since it's test code I don't really care haha